### PR TITLE
fix get users bug and regex

### PR DESCRIPTION
**Pull request by ndoriomedov**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/917
+fix database request

### CODE REVIEW

**Review by MariyaSin — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public class CreateUserRequestValidator : AbstractValidator<CreateUserRequest>,
     private static Regex NumberRegex = new(@"\d");
     private static Regex SpecialCharactersRegex = new(@"[$&+,:;=?@#|<>.^*()%!]");
     private static Regex SpaceRegex = new(@"^[^@\s]*$");
-    private static Regex NameRegex = new(@"^[a-zA-Zа-яА-ЯёЁ'][a-zA-Z-а-яА-ЯёЁ' ]+[a-zA-Zа-яА-ЯёЁ']?$");
+    private static Regex NameRegex = new(@"^([a-zA-Zа-яА-ЯёЁ]+|[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+|[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+)$");
```
Comment (BelehovEgor) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:16]: why do you duplicate same parts?

Comment (ndoriomedov) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs:16]: the first one before "|" symbol is for one-letter names/surnames
the second ([a-zA-Zа-яА-ЯёЁ]+[-|']?[a-zA-Zа-яА-ЯёЁ]+) is for names with one ' or - symbol (those names must cintain at least two letters - in the start and the end of the word)
similarly the third is for words containing  3-letters and 2 ' or -

File: src/UserService.Broker/Consumers/GetUsersDataConsumer.cs [hunk 2]
```diff
@@ -30,12 +30,12 @@ public class GetUsersDataConsumer : IConsumer<IGetUsersDataRequest>
...
     private async Task<List<UserData>> GetUserInfoAsync(IGetUsersDataRequest request)
     {
-      List<(DbUser user, Guid avatarId)> users = await _userRepository.GetWithAvatarsAsync(request.UserIds);
+      List<(DbUser user, Guid? avatarId)> users = await _userRepository.GetWithAvatarsAsync(request.UserIds);
 
       return users
         .Select(users => new UserData(
           users.user.Id,
-          users.avatarId == default ? null : users.avatarId,
+          users.avatarId.HasValue ? users.avatarId.Value : null,
```
[the remark this answers is not in the document]
Comment (ndoriomedov) [reply src/UserService.Broker/Consumers/GetUsersDataConsumer.cs]: users is List<(DbUser user, Guid? avatarId)> - the collection of users and their avatars

[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/UserService.Broker/Consumers/GetUsersDataConsumer.cs]: I told about the name of the variable in a lambda expression.

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs:16]: ok, I couldn't do much better

**Review by BelehovEgor — APPROVED** [review]

File: src/UserService.Broker/Consumers/GetUsersDataConsumer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndoriomedov) [reply src/UserService.Broker/Consumers/GetUsersDataConsumer.cs]: renamed to userData

**Pull request merged**